src/util/triemap/triemap.c

```diff
@@ -492,6 +492,11 @@ void* TrieMap_DONT_CARE_REPLACE(void *oldval, void *newval) {
   return newval;
 }
 
+void* TrieMap_NOP_REPLACE(void *oldval, void *newval) {
```

Comment:
We already have TrieMap_DONT_CARE_REPLACE
See if we can combine the two

Reply:
DONT_CARE_REPLACE and NOP_REPLACE actually have opposite behaviors, in that the former always replaces the held value, and the latter never does.

For stores, we'll only use DONT_CARE when introducing indices (the only real value in those tries). All the other times we should use NOP, so that if we re-insert an a property (like we do with every SET expression), we don't risk accidentally losing an index pointer.

The naming is a little confusing, but we also use a NOP free callback in the triemap, so I'm not quite sure how to improve it.